Sources/TensorFlow/Operators/LinearAlgebra.swift

```diff
@@ -177,6 +177,25 @@ public func trace<T: TensorFlowNumeric>(_ matrix: Tensor<T>) -> Tensor<T> {
     return matrix.diagonalPart().sum(squeezingAxes: -1)
 }
 
+/// Computes the determinant of an optionally batched matrix.
+/// 
+/// - Parameter matrix: A tensor of shape `[..., M, M]`.
+/// - Returns: A tensor containing the determinants of all input submatrices.
+@inlinable 
+func det<T: TensorFlowFloatingPoint>(_ matrix: Tensor<T>) -> Tensor<T> {
+    _Raw.matrixDeterminant(matrix)
+}
+
+/// Computes the sign and the natural logarithm of the absolute value of the determinant of an
+/// optionally batched square matrix.
+///
+/// - Parameter matrix: A tensor of shape `[..., N, M, M]`.
+/// - Returns: A tensor containing the determinants of all input submatrices.
+@inlinable
+func slogdet<T: TensorFlowFloatingPoint>(_ matrix: Tensor<T>) -> (Tensor<T>, Tensor<T>) {
```

Comment:
Is it possible to define the derivative of `slogdet`?

---

Motivation: `slogdet` overlaps with the previously defined `func logdet` below:

```swift
@differentiable(wrt: matrix where T: TensorFlowFloatingPoint)
func logdet<T: TensorFlowFloatingPoint>(_ matrix: Tensor<T>) -> Tensor<T> {
    return 2.0 * log(cholesky(matrix).diagonalPart()).sum(squeezingAxes: -1) 
}
```

(Aside: `func logdet` should be renamed to `func logDeterminant`.)

`func logdet` is differentiable, while `func slogdet` is not. If we can define a derivative for `func slogdet`, we could consider changing the implementation of `func logdet` to call `func slogdet` instead of custom computation.

Reply:
cc @param087, who implemented `func logdet` in https://github.com/tensorflow/swift-apis/pull/592.
I wonder what you used as the reference implementation for `func logdet`? I'm curious why it uses custom logic instead of forwarding to a `_Raw` op.

Reply:
@Shashi456: to make progress on this, could you please rename the existing `func logdet` to `func logDeterminant`? We can try to unify or code share `func logDeterminant` and `func signAndLogDeterminant` in a follow-up.

Reply:
@dan-zheng I would recommend against this renaming. `slogdet` and `logdet` are well known and used by every single scientific language I know - Python (numpy), R, Julia, Matlab. The `det` abbreviation is also broadly used in the literature as well.

Reply:
@Shashi456 ^^^

Reply:
@dan-zheng since I mostly take inspiration from the tensorflow-python API and check numpy linear algebra library for additional consistency. I initially named the functions slogdet and det.

Reply:
I feel reusing existing term-of-art names here is reasonable. Could you please add `(sign:logDeterminant:)` result tuple element labels to `slogdet` please?

```swift
func slogdet<T: TensorFlowFloatingPoint>(_ matrix: Tensor<T>) -> (sign: Tensor<T>, logDeterminant: Tensor<T>) {
```

Reply:
> cc @param087, who implemented `func logdet` in #592.
> I wonder what you used as the reference implementation for `func logdet`? I'm curious why it uses custom logic instead of forwarding to a `_Raw` op.

@dan-zheng, I had taken reference from https://github.com/tensorflow/tensorflow/blob/r2.1/tensorflow/python/ops/linalg/linalg_impl.py#L72-L103 to implement `logdet`.  Considering the python implementation of `logdet` and making it differentiable, I found custom logic is more promising than that of forwarding to a `_Raw` op.